src/data-workspace/data-details-sidebar/comment.js

```diff
@@ -13,21 +13,13 @@ const errorMessage = i18n.t(
     'There was a problem loading the comment for this data item'
 )
 
-function CommentEditForm({ item, open, setOpen, syncComment, onCancel }) {
-    const [{ dataSetId: ds, periodId: pe, orgUnitId: ou }] =
-        useContextSelection()
-
+function CommentEditForm({ item, open, setOpen, syncComment, closeEditor }) {
     const onSubmit = (values) => {
-        const variables = {
-            ds,
-            ou,
-            pe,
-            co: item.categoryOptionCombo,
-            de: item.dataElement,
-            comment: values.comment,
-        }
-
-        return syncComment(variables)
+        return syncComment(
+            // Don't send `undefined` (or 'undefined' will be stored as the comment)
+            { comment: values.comment || '' },
+            { onSuccess: closeEditor }
```

Comment:
do we need to have the editor close onMutate? In offline mode, it seems it won't close until the mutation goes through, which I assume is not what we want?

Reply:
I think you're right, good point!